### PR TITLE
Ensure routes to items are encoded correctly

### LINKED ISSUE #19052: Show a link to the item opens /items/directus_users instead of User Directory
### Describe the Bug

Show a link to the item in One to Many relation ship, forwards user to /items/directus_users/:id, instead of /users/:id. Because of that, user can see unnecessary fields, like Last Page and Last Access 

### To Reproduce

Create one to many relationship to directus_users model with Show a link to the item setting enabled. Click on that link

### Directus Version

v10.4.2

### Hosting Strategy

Self-Hosted (Custom)

### LINKED ISSUE #19603: Kanban layout # character in ID is not url encoded
### Describe the Bug

In the Kanban layout, when clicking on an item with an ID containing a hash symbol, the hash is not properly encoded, so the subsequent navigation fails.

### To Reproduce

Create item with # character in ID.
Open Kanban layout
Click item.
Note in chrome dev tools that API call is made, but only the ID up until the hash is used, resulting in a "page not found error"
furthermore, if # character is the first character, the navigation doesn't work at all.

https://github.com/directus/directus/assets/21194068/2bff52cd-1f2b-478d-910a-a2a76bc7ae79

### Directus Version

10.5.3

### Hosting Strategy

Directus Cloud

### LINKED ISSUE #19605: Tabular layout, item with # in ID in bookmark
### Describe the Bug

Some items have # characters in the primary key field.
When a bookmark search is used and an item in the table layout is clicked, the app errors out.

The error is here: https://github.com/directus/directus/blob/ece722871b2063c43f8833e73928d4a280041a7a/app/src/layouts/tabular/index.ts#L308C19-L308C19

### To Reproduce

Create collection with manual text input for ID.
Add some items with IDs including a # symbol
Create a bookmark with the tabular layout.

Note the Directus url now include querystring "?bookmark=xyz"
Note that when you click on an item with a # in the ID, the app becomes unresponsive.

https://github.com/directus/directus/assets/21194068/3d61b2e5-d2ad-4822-94f2-48a50a6d587a

### Directus Version

10.5.3

### Hosting Strategy

Directus Cloud

### PR DESCRIPTION
**Pull request by alexvdvalk**
Ensure routes to items are encoded correctly for primary keys with special characters, by using the same `get-route` utils everywhere.

Fixes #19603, fixes #19605, fixes #19052

### CODE REVIEW

**Review by paescuj — APPROVED** [review]
Nice 🚀

File: .changeset/tasty-zoos-repeat.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@directus/app": patch
+---
+
+Ensured links to items in Kanban and Calendar layouts are properly encoded
```
Comment (paescuj) [inline .changeset/tasty-zoos-repeat.md:5]:
```suggestion
Ensured links to items with special characters as primary key are properly encoded
```

File: app/src/utils/get-route.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+	if (collection === null) return '';
+
+	const collectionRoute = getCollectionRoute(collection);
+
+	if (collectionRoute === '') return '';
+
+	if (isAccessibleSystemCollection(collection) && 'singleton' in accessibleSystemCollections[collection])
+		return collectionRoute;
+
+	const itemRoute = primaryKey === '+' ? '+' : encodeURIComponent(primaryKey);
```
Comment (azrikahar) [inline app/src/utils/get-route.ts:66]:
```suggestion
	const itemRoute = primaryKey === '+' ? primaryKey : encodeURIComponent(primaryKey);
```

(this change really isn't necessary, but just a subjective tweak!)

File: app/src/utils/get-route.test.ts [hunk 3]
```diff
@@ -0,0 +1,83 @@
+import { getCollectionRoute, getItemRoute, getSystemCollectionRoute } from '@/utils/get-route';
+import { describe, expect, it, vi } from 'vitest';
```
Comment (azrikahar) [inline app/src/utils/get-route.test.ts:2]:
```suggestion
import { describe, expect, it } from 'vitest';
```

lint tweak

**Review by azrikahar — APPROVED** [review]
LGTM! I also believe the follow up change to list-o2m (and list-m2m) subsequently resolves #19052, and confirmed it on my end so I'll mark it as resolved by this PR before merging 👍

File: app/src/modules/content/index.ts [hunk 4]
```diff
@@ -1,103 +1,32 @@
...
-		}
-	}
-
-	if (to.params.collection === 'directus_webhooks') {
-		if (to.params.primaryKey) {
-			return `/settings/webhooks/${to.params.primaryKey}`;
-		} else {
-			return '/settings/webhooks';
-		}
-	}
```
[the remark this answers is not in the document]
Comment (paescuj) [reply app/src/modules/content/index.ts:94]: Haha, yes 😅 🧹

**Pull request merged**